pkg/storage/fs/cephmount/cephmount.go

```diff
@@ -1258,6 +1323,289 @@ func (fs *cephmountfs) Unlock(ctx context.Context, ref *provider.Reference, lock
+		userId := grant.Grantee.GetUserId()
+		if userId == nil {
+			return errors.New("cephmount: user grantee without user ID")
+		}
+		identifier = userId.OpaqueId
+
+		// Resolve username to UID for setfacl (OpaqueId is always a username)
+		userInfo, err := user.Lookup(identifier)
+		if err != nil {
+			return errors.Errorf("cephmount: user '%s' does not exist in /etc/passwd. "+
+				"All users must be available on the local system. Original error: %v", identifier, err)
+		}
+		identifier = userInfo.Uid
+		log.Debug().
+			Str("username", userId.OpaqueId).
+			Str("uid", identifier).
+			Msg("cephmount: resolved username to UID")
+
+		aclEntry = fmt.Sprintf("u:%s:%s", identifier, fs.permissionsToACLString(grant.Permissions))
+
+	case provider.GranteeType_GRANTEE_TYPE_GROUP:
+		groupId := grant.Grantee.GetGroupId()
+		if groupId == nil {
+			return errors.New("cephmount: group grantee without group ID")
+		}
+		identifier = groupId.OpaqueId
+
+		// Resolve group name to GID for setfacl (OpaqueId is always a groupname)
+		groupInfo, err := user.LookupGroup(identifier)
+		if err != nil {
+			return errors.Errorf("cephmount: group '%s' does not exist in /etc/group. "+
+				"All groups must be available on the local system. Original error: %v", identifier, err)
+		}
+		identifier = groupInfo.Gid
+		log.Debug().
+			Str("groupname", groupId.OpaqueId).
+			Str("gid", identifier).
+			Msg("cephmount: resolved groupname to GID")
+
+		aclEntry = fmt.Sprintf("g:%s:%s", identifier, fs.permissionsToACLString(grant.Permissions))
+
+	default:
+		return errors.New("cephmount: invalid grantee type")
+	}
+
+	// Build setfacl command
+	args := []string{"-m", aclEntry}
+	if info.IsDir() {
+		// Also set default ACLs for directories
+		args = append(args, "-m", "d:"+aclEntry)
+		// Recursive flag must come last before the path
+		args = append(args, "-R")
+	}
+	args = append(args, fullPath)
+
+	log.Debug().
+		Str("path", fullPath).
+		Str("acl_entry", aclEntry).
+		Strs("args", args).
+		Msg("cephmount: executing setfacl")
+
+	// Execute setfacl
+	cmd := exec.CommandContext(ctx, "setfacl", args...)
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Error().
+			Err(err).
+			Str("path", fullPath).
+			Str("output", string(output)).
+			Msg("cephmount: setfacl failed")
+		return errors.Wrapf(err, "cephmount: setfacl failed: %s", string(output))
+	}
+
+	log.Debug().
+		Str("path", fullPath).
+		Msg("cephmount: setfacl succeeded")
+
+	return nil
+}
+
+// removeGrantViaSetfacl removes a grant using the setfacl system command
+func (fs *cephmountfs) removeGrantViaSetfacl(ctx context.Context, path string, grant *provider.Grant) error {
+	log := appctx.GetLogger(ctx)
+	fullPath := filepath.Join(fs.chrootDir, path)
+
+	// Check if it's a directory for recursive flag
+	info, err := os.Stat(fullPath)
+	if err != nil {
+		return errors.Wrap(err, "cephmount: failed to stat path")
+	}
+
+	// Build ACL removal entry string based on grantee type
+	var aclEntry string
```

Comment:
Can you not factor out the acl building (since you do this for both the add and remove grant)?